src/resources/views/buttons/delete.blade.php

```diff
@@ -1,52 +1,55 @@
 
 <script>
-	if (typeof registerDeleteButtonAction != 'function') {
-	  function registerDeleteButtonAction() {
-        $("[data-button-type=delete]").unbind('click');
-        // CRUD Delete
-        // ask for confirmation before deleting an item
-        $("[data-button-type=delete]").click(function(e) {
-          e.preventDefault();
-          var delete_button = $(this);
-          var delete_url = $(this).attr('href');
+	if (typeof deleteEntry != 'function') {
+	  $("[data-button-type=delete]").unbind('click');
 
-          if (confirm("{{ trans('backpack::crud.delete_confirm') }}") == true) {
-              $.ajax({
-                  url: delete_url,
-                  type: 'DELETE',
-                  success: function(result) {
-                      // Show an alert with the result
-                      new PNotify({
-                          title: "{{ trans('backpack::crud.delete_confirmation_title') }}",
-                          text: "{{ trans('backpack::crud.delete_confirmation_message') }}",
-                          type: "success"
-                      });
-                      // delete the row from the table
-                      delete_button.parentsUntil('tr').parent().remove();
-                  },
-                  error: function(result) {
-                      // Show an alert with the result
-                      new PNotify({
-                          title: "{{ trans('backpack::crud.delete_confirmation_not_title') }}",
-                          text: "{{ trans('backpack::crud.delete_confirmation_not_message') }}",
-                          type: "warning"
-                      });
-                  }
-              });
-          } else {
-              new PNotify({
-                  title: "{{ trans('backpack::crud.delete_confirmation_not_deleted_title') }}",
-                  text: "{{ trans('backpack::crud.delete_confirmation_not_deleted_message') }}",
-                  type: "info"
-              });
-          }
-        });
+	  function deleteEntry(button) {
+	      // ask for confirmation before deleting an item
+	      // e.preventDefault();
+	      var button = $(button);
+	      var route = button.attr('data-route');
+	      var row = $("#crudTable a[data-route='"+route+"']").parentsUntil('tr').parent();
+
+	      if (confirm("{{ trans('backpack::crud.delete_confirm') }}") == true) {
+	          $.ajax({
+	              url: route,
+	              type: 'DELETE',
+	              success: function(result) {
+	                  // Show an alert with the result
+	                  new PNotify({
+	                      title: "{{ trans('backpack::crud.delete_confirmation_title') }}",
+	                      text: "{{ trans('backpack::crud.delete_confirmation_message') }}",
+	                      type: "success"
+	                  });
+
+	                  // close the modal, if any
+	                  $('.modal').modal('hide');
+
+	                  // delete the row from the table
```

Comment:
"Remove" (v) "Add" -- is this a redundant comment?

Reply:
Right, changed, thanks!